lib/modules/manager/gradle/artifacts.ts

```diff
@@ -0,0 +1,150 @@
+import is from '@sindresorhus/is';
+import { quote } from 'shlex';
+import { dirname } from 'upath';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import { findUpLocal, readLocalFile, writeLocalFile } from '../../../util/fs';
+import { getFileList, getRepoStatus } from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import {
+  extraEnv,
+  getJavaConstraint,
+  gradleWrapperFileName,
+} from '../gradle-wrapper/utils';
+import { getLockFileContentMap } from '../nuget/artifacts';
```

Comment:
😯 we probably should extract that function to some common manager file.

Reply:
Do you have an existing file in mind where this would fit? Or is this just meant as a note for a later refactoring PR?

Reply:
Looks like it could live in util, maybe util/fs, perhaps with some renaming too. Doesn't seem particularly specific to lock files itself